### PR TITLE
Add majorVersion option to ls-remote

### LINKED ISSUE #188: Support arguments to ls-remote
Basically, having an equivalent of `nvm ls-remote [parameter]` would be cool. In particular, I often invoke `nvm ls-remote 12` which only shows `12.x.y` versions; with fnm I can only list all versions from which I then need to grep as the list is quite long.

### PR DESCRIPTION
**Pull request by tatchi**
My attempt to fix #188 :)

To try it:

```bash
fnm ls-remote 12 #or another version
```

### CODE REVIEW

**Comment by Schniz** [comment]
Also, would you mind adding a feature test? 😄

File: executable/ListRemote.re [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
 
-       Console.log(<Pastel ?color> str </Pastel>);
-     });
+  switch (versions) {
+  | [] =>
+    Console.log(
+      <Pastel color=Pastel.Red>
+        "No versions found that match your criterias."
+      </Pastel>,
+    )
```
[the remark this answers is not in the document]
Comment (tatchi) [reply executable/ListRemote.re]: good point 👍

File: executable/FnmApp.re [hunk 2]
```diff
@@ -14,7 +14,7 @@ module Commands = {
   let use = (version, quiet) => Use.run(~version, ~quiet) |> runCmd;
   let alias = (version, name) => Alias.run(~name, ~version) |> runCmd;
   let default = version => Alias.run(~name="default", ~version) |> runCmd;
-  let listRemote = () => ListRemote.run() |> runCmd;
+  let listRemote = majorVersion => ListRemote.run(~majorVersion) |> runCmd;
```
[the remark this answers is not in the document]
Comment (tatchi) [reply executable/FnmApp.re]: Indeed that would be more flexible :)

**Comment by tatchi** [comment]
@Schniz I've addressed your comments and added a few tests :)

File: test/TestListRemote.re [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      allVersions6_11 |> List.map(formatVersion),
+    );
+  });
+  test(
+    "Should display remote version matching a specific version", ({expect}) => {
+    let result = run([|"ls-remote", "6.11.3"|]);
+
+    let versions =
+      result
+      |> String.split_on_char('\n')
```
Comment (Schniz) [inline test/TestListRemote.re]: consider extracting a helper `filterVersionNumbers` for the repetitive filtering done in the following lines

You can also make it skip the `* ` on each line, making the comparison easier to understand (`expect.lines(versions).toEqualLines(allVersions)`)

File: test/TestListRemote.re [hunk 4]
```diff
@@ -0,0 +1,138 @@
+open TestFramework;
+
+describe("List Remote", ({test}) => {
+  let allVersions6 = [
```
Comment (Schniz) [inline test/TestListRemote.re]: what do you think about moving this into the test scope itself?

File: test/TestListRemote.re (hunk 3 above)
Comment (tatchi) [reply test/TestListRemote.re]: Indeed there's quite a few repetitions there :P

File: test/TestListRemote.re (hunk 4 above)
Comment (tatchi) [reply test/TestListRemote.re]: Done :)

**Comment by Schniz** [comment]
can you retrigger everything? maybe by pushing an empty commit?

**Comment by tatchi** [comment]
> can you retrigger everything? maybe by pushing an empty commit?

Can you try this: https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers ?

/azp run

**Comment by Schniz** [comment]
I renamed my Azure's namespace 🤔  maybe this is the problem

**Comment by Schniz** [comment]
Thanks @tatchi!

**Pull request merged**